packages/backend/src/apiKey/service.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseService } from '../shared/baseService'
+import { SessionService } from '../session/service'
+import { ApiKey } from './model'
+import { uuid } from '../connector/core'
+import bcrypt from 'bcrypt'
+import { Transaction } from 'knex'
+import { SessionKey } from '../session/util'
+import { ApiKeyError } from './errors'
+
+export interface ApiKeyService {
+  create(apiKey: ApiKeyOptions, trx?: Transaction): Promise<NewApiKey>
+  get(apiKey: ApiKeyOptions): Promise<ApiKey[]>
+  redeem(sessionKey: SessionKeyOptions): Promise<SessionKey | ApiKeyError>
+  deleteAll(apiKey: ApiKeyOptions, trx?: Transaction): Promise<void>
+}
+
+interface ServiceDependencies extends BaseService {
+  sessionService: SessionService
+}
+
+type ApiKeyOptions = {
+  accountId: string
+}
+
+type SessionKeyOptions = {
+  accountId: string
+  key: string
+}
+
+interface NewApiKey extends ApiKey {
+  key: string
+}
+
+export async function createApiKeyService({
+  logger,
+  knex,
+  sessionService: sessionKeyService
+}: ServiceDependencies): Promise<ApiKeyService> {
+  const log = logger.child({
+    service: 'ApiKeyService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex,
+    sessionService: sessionKeyService
+  }
+  return {
+    create: (options, trx) => createApiKey(deps, options, trx),
+    get: (options) => getApiKeys(deps, options),
+    redeem: (options) => redeemSessionKey(deps, options),
```

Comment:
```suggestion
    redeem: (options) => redeemApiKey(deps, options),
```

Reply:
I just looked up the meaning of "redeem" and it looks like I had it backwards. Thanks!